### PR TITLE
LPS-127201 Support Social Traffic Detail

### PR DESCRIPTION
**Pull request by cgoncas**
**Motivation**

The Social-page-referrers endpoint is now available in AC (https://issues.liferay.com/browse/LRAC-5747). So we need to update Tango code in order to call this endpoint and obtain all the information.

**How to test it**

Integration tests have been added to this pull.

_Real Tests_

1. Add the next JS script to the file liferay-portal/modules/apps/analytics/analytics-web/src/main/resources/META-INF/resources/dynamic_include/top_head.jsp

`	Object.defineProperty(document, 'referrer', {
		configurable: true,
		get: function() {
			return "facebook.com";
		}
	});`
	
2. Deploy analytics-web module
3. Go to the page you want to test with a user different than test/test and a URL different than localhost:8080
4. Check that the right referrer is sent to AC

Screenshot 2021-02-09 at 18 12 35

5. Wait 24h
5. Go to the Control Performance Panel of the Page and check that the Social graph has a link

### CODE REVIEW

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/data/provider/AnalyticsReportsDataProvider.java [hunk 1]
```diff
@@ -152,6 +153,42 @@ public HistoricalMetric getHistoricalViewsHistoricalMetric(
...
+			return entriesStream.map(
+				entry -> new ReferringSocialMedia(
+					entry.getKey(), Math.toIntExact(entry.getValue()))
+			).collect(
+				Collectors.toList()
+			);
+		}
+		catch (Exception exception) {
+			throw new PortalException(
+				"Unable to get acquisition channels", exception);
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/data/provider/AnalyticsReportsDataProvider.java:188]: It should be social media related, right?

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/data/provider/AnalyticsReportsDataProvider.java [hunk 2]
```diff
@@ -152,6 +153,42 @@ public HistoricalMetric getHistoricalViewsHistoricalMetric(
...
+
+		try {
+			String response = _asahFaroBackendClient.doGet(
+				companyId,
+				"api/1.0/pages/social-page-referrers?canonicalURL=" + url +
+					"&interval=D&rangeKey=30");
+
+			TypeFactory typeFactory = _objectMapper.getTypeFactory();
+
+			Map<String, Long> socialTrafficChannels = _objectMapper.readValue(
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/data/provider/AnalyticsReportsDataProvider.java:168]: This name sounds a bit estrange for me, What do you think about `socialPageReferrers`?

File: modules/dxp/apps/analytics/analytics-reports-web/src/test/java/com/liferay/analytics/reports/web/internal/data/provider/AnalyticsReportsDataProviderTest.java [hunk 3]
```diff
@@ -170,6 +171,39 @@ public void testGetHistoricalReadsHistoricalMetric() throws Exception {
 		Assert.assertEquals(localDate, zonedDateTime.toLocalDate());
 	}
 
+	@Test
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-web/src/test/java/com/liferay/analytics/reports/web/internal/data/provider/AnalyticsReportsDataProviderTest.java:175]: ❤️

File: modules/dxp/apps/analytics/analytics-reports-web/src/test/java/com/liferay/analytics/reports/web/internal/data/provider/AnalyticsReportsDataProviderTest.java [hunk 4]
```diff
@@ -170,6 +171,39 @@ public void testGetHistoricalReadsHistoricalMetric() throws Exception {
...
+	public void testGetReferringSocialMediaList() throws Exception {
+		AnalyticsReportsDataProvider analyticsReportsDataProvider =
+			new AnalyticsReportsDataProvider(
+				_getHttp(
+					Collections.singletonMap(
+						"/social-page-referrers",
+						JSONUtil.put(
+							"direct", 6.0
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-web/src/test/java/com/liferay/analytics/reports/web/internal/data/provider/AnalyticsReportsDataProviderTest.java:183]: This is not important, would it be better to use a real social media instead?

**Review by darquesdev — APPROVED** [review]
Beautiful PR!! just left minor comments

**Review by dgarciasarai — COMMENTED** [review]
Just starting reviewing :)

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
> Beautiful PR!! just left minor comments

Thanks a lot for the review. I have applied all the feedback you have proposed :)

**Comment by dgarciasarai** [comment]
Hey @cgoncas, I am removing the TimeSpanSelector component inside SocialDetail file. Frontend will need `referringSocialMedia` with `name`, `title`, `trafficAmount` to display the socials. Is that correct? If not, please share with me the data schema for social traffic details. Thanks!

**Comment by cgoncas** [comment]
> SocialDetail

Hey @dgarciasarai !!

I think that if you just hide the TimeSpanSelector is OK.

I wouldn't delete it, as this is a task we will do in the future, you can still send the data but I won't take that into account.

Thanks!!!

**Comment by dgarciasarai** [comment]
> Hey @dgarciasarai !!
> 
> I think that if you just hide the TimeSpanSelector is OK.
> 
> I wouldn't delete it, as this is a task we will do in the future, you can still send the data but I won't take that into account.
> 
> Thanks!!!

Ok sure! You're right, I'll hide it :) 

Can you provide a social schema to try the UI with mock data? Thanks :)

**Comment by cgoncas** [comment]
> > Hey @dgarciasarai !!
> > I think that if you just hide the TimeSpanSelector is OK.
> > I wouldn't delete it, as this is a task we will do in the future, you can still send the data but I won't take that into account.
> > Thanks!!!
> 
> Ok sure! You're right, I'll hide it :)
> 
> Can you provide a social schema to try the UI with mock data? Thanks :)

Hey @dgarciasarai !!

You can use our old mocks (https://github.com/liferay-tango/liferay-portal/pull/674), the old data should work exactly the same as we did before, only the way we fill the info in the backend has changed.

Thanks!!

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/model/ReferringSocialMedia.java [hunk 5]
```diff
@@ -61,6 +61,15 @@ public JSONObject toJSONObject(ResourceBundle resourceBundle) {
 		);
 	}
 
+	@Override
```
Comment (dgarciasarai) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/model/ReferringSocialMedia.java:64]: Hey @cgoncas, here we don't have title, as in the mocks (https://github.com/liferay-tango/liferay-portal/pull/674), is there any reason? 🤔

Comment (cgoncas) [reply modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/model/ReferringSocialMedia.java:64]: Hey @dgarciasarai !!!

The toString method is not related whit the info you will receive in the frontend.

Thanks!!

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:forward

**Pull request closed without merging**